### PR TITLE
Impac! permissions

### PR DESCRIPTION
**Pull request by cesar-tonnoir**
- [x] add dependency to MnoHub: https://github.com/maestrano/maestrano-hub/pull/818 and Mnoe: https://github.com/maestrano/mno-enterprise/pull/588

- Rework conditional display of Impac! and menu buttons based on ACL for user / selected organization
- Add `ImpacConfigSvc.configure` = ability to pass a function that customises the list of organizations before passing them to Impac!

**Note**: The ACL for a tenant should be modified via ActiveAdmin (tenant metadata) instead of using the `ImpacConfigSvc.configure` helper. The later is kept in the code as an improvement if some overrides are needed on the organizations passed to Impac!

### CODE REVIEW

File: src/app/views/apps/dashboard-apps-list.controller.coffee [hunk 1]
```diff
@@ -104,11 +104,19 @@ angular.module 'mnoEnterpriseAngular'
...
-          $scope.isLoading = true
-          MnoeAppInstances.getAppInstances().then(
-            ->
-              $scope.isLoading = false
-              $scope.apps = MnoeAppInstances.appInstances
-          )
+        $scope.isLoading = true
+        ImpacConfigSvc.getOrganizations().then(
+          (resp) ->
+            selectedOrg = _.find(resp.organizations, { id: parseInt(val) })
```
Comment (alexnoox) [inline src/app/views/apps/dashboard-apps-list.controller.coffee]: FYI `parseInt` will need to be removed in branch `2.0`

File: src/app/components/impac-config/impac-config.svc.js.coffee [hunk 2]
```diff
@@ -1,6 +1,14 @@
 angular.module 'mnoEnterpriseAngular'
   .service('ImpacConfigSvc' , ($log, $q, MnoeCurrentUser, MnoeOrganizations) ->
 
+    _self = @
+
+    @config =
+      overrideOrgs: (orgs) -> orgs
```
[the remark this answers is not in the document]
Comment (cesar-tonnoir) [reply src/app/components/impac-config/impac-config.svc.js.coffee]: It's meant to be overriden when you configure Impac!: if you need to add / remove attributes to / from the organizations objects...

File: src/app/impac.config.coffee [hunk 3]
```diff
@@ -56,6 +56,21 @@ angular.module 'mnoEnterpriseAngular'
...
+  #     angular.merge(org, {
+  #       acl:
+  #         related:
+  #           impac: { show: true }
+  #           dashboards: { update: true }
+  #           widgets: { update: true }
+  #           kpis: { show: true }
+  #     })
+  #   )
+  # ImpacConfigSvc.configure(overrideAcl)
```
[the remark this answers is not in the document]
Comment (cesar-tonnoir) [reply src/app/impac.config.coffee]: It wouldn't make sense to pass an object: this is meant to perform customizations (on the organizations) that cannot be predicted in advance?..

File: src/app/views/apps/dashboard-apps-list.controller.coffee (hunk 1 above)
Comment (cesar-tonnoir) [reply src/app/views/apps/dashboard-apps-list.controller.coffee]: ok - what do you use instead?

**Comment by cesar-tonnoir** [comment]
@alexnoox can you plz approve?

File: src/app/views/apps/dashboard-apps-list.controller.coffee (hunk 1 above)
Comment (alexnoox) [reply src/app/views/apps/dashboard-apps-list.controller.coffee]: in `2.0`, ids are typed as string

**Pull request merged**